Framework/Geometry/src/Instrument.cpp

```diff
@@ -1014,7 +1014,18 @@ void Instrument::saveNexus(::NeXus::File *file,
 
   // Now the parameter map, as a NXnote via its saveNexus method
   if (isParametrized()) {
-    const Geometry::ParameterMap &params = *getParameterMap();
+    Geometry::ParameterMap params(*getParameterMap());
+    // Masking is in DetectorInfo. Insert it into ParameterMap so it is saved
+    // alongside other parameters.
+    if (m_detectorInfo) {
```

Comment:
I did wonder if that was the reason that you did it. It would indeed be better if it didn't depend on legacy things. Sounds like you've thought about it in any case.